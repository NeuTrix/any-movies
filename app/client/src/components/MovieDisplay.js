import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography'


// include props declartaions (classes)
const propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  movie: PropTypes.instanceOf(Object).isRequired,
  posterUrl: PropTypes.instanceOf(Object).isRequired,
}

function MovieDisplay(props) {

  const { classes, movie, posterUrl } = props
  
  // if (movie.Ratings) listed, show them
  const ratings = movie.Ratings && movie.Ratings.map((rating, index) => {
    return ( 
      <div key={index} className={classes.ratingUnit}> 
        <div style={{gridArea: 'critic'}} > {rating.Source}: </div>
        <div style={{gridArea: 'grade', textAlign: 'right'}} > {rating.Value} </div>
      </div> 
    )
  })

  // if actors listed, make a list
  const actors = movie.Actors && movie.Actors.split(',').map((actor, index) => {
    return ( <p key={index}> - {actor} </p> )
  })

  return (
    <div className={classes.main} >

      <div className={classes.title}>
        <Typography variant="h4"> { movie.Title } </Typography> 
      </div>

      <div className={classes.image} >
        <img className={classes.poster} src={posterUrl} alt="movie poster"/>
        <div> Released: {movie.Year} </div> <br/>
        <div> Rated 
          <h3>{movie.Rated} </h3> 
        </div> 
        <div> <h6> imdbID: { movie.imdbID }</h6> </div>
      </div>

      <div className={classes.info}>

        <div>
          <h4> Genre: </h4>
          <p> - {movie.Genre} </p>
        </div> 

        <div> 
          <h4> Director: </h4>
          <p> - {movie.Director} </p> 
        </div> 
       
        <div> 
          <h4> Starring: </h4>
          <p> { actors } </p>
        </div> 
       
      </div>

      <div className={classes.ratings} > 
          <div> 
            <h4> Critics Ratings: </h4>
            <p> { ratings } </p> 
          </div>
      </div>
      
      <div className={classes.plot} > 
        <h4> Movie Plot: </h4>
        <div> {movie.Plot} </div> 
      </div> 

    </div>
  )
}

const styles = theme => ({
  main: {
    border: '1px solid lightgrey',
    display: 'inline-grid',
    gridTemplateAreas: `
      "title title"
      "image info"
      "plot plot"
      "ratings ratings"
    `,
    // gridTemplateAreas: `
    //   "title title"
    //   "poster rated"
    //   "poster year"
    //   "poster mrate"
    //   "poster desr"
    // `,
    gridTemplateColumns: '2fr 2fr',
  },

  title: {
    gridArea: 'title',
    padding: theme.spacing.unit,
    borderBottom: '1px solid lightgrey',
  },

  info: {
    textAlign: 'left',
    gridArea: 'info',
    padding: theme.spacing.unit * 2 ,
  },

  image: {
    gridArea: 'image',
    padding: theme.spacing.unit,
    borderRight: '1px solid lightgrey',
  },

  poster: {
    maxWidth: 175,
    padding: theme.spacing.unit,
  },

  plot: {
    gridArea: 'plot',
    border: '1px solid lightgrey',
    padding: theme.spacing.unit,
    textAlign: 'left',
  },

  ratings: {
    gridArea: 'ratings',
    textAlign: 'left', 
    padding: theme.spacing.unit,
  },

  ratingUnit: {
    display: 'grid',
    gridTemplateAreas:`
      "critic grade"
    `,
    gridTemplateColumns: '3rf 1fr',
    // textAlign:'left',
  }

})

export default withStyles(styles)(MovieDisplay)
