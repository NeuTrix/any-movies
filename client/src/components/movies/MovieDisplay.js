import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { Button } from '@material-ui/core';
import FavoriteTwoTone from '@material-ui/icons/FavoriteTwoTone';

const propTypes = {
	classes: PropTypes.instanceOf(Object).isRequired,
	curr_movie: PropTypes.instanceOf(Object).isRequired,
	favouriteMovie: PropTypes.func.isRequired, // add to favourites
};

function MovieDisplay(props) {
	const { classes, curr_movie } = props;

	// add to favourites
	const onClick = (e) => {
		e.preventDefault();
		props.favouriteMovie();
	};

	// generate list of movie ratings
	const ratings = curr_movie.Ratings && curr_movie.Ratings.map((rating) => (
		<div key={curr_movie.imdbID} className={classes.ratingUnit}>
			<div style={{ gridArea: 'critic' }}>
				{`${rating.Source} :`}
			</div>
			<div style={{ gridArea: 'grade', textAlign: 'right' }}>
				{rating.Value}
			</div>
		</div>
	));

	// if actors listed, make a list
	const actors = curr_movie.Actors && curr_movie.Actors.split(',')
		.map((actor) => ( <div key={curr_movie.imdbID}> - {actor} </div> )
	);

	return (
		<div className={classes.main}>
			
			<div className={classes.titlebar}>
				<div className={classes.fav}>
					<Button 
						variant="fav"
						onClick={onClick}
					>
						<FavoriteTwoTone color="primary" />
					</Button>
				</div>

				<div className={classes.title} >
					<Typography variant="h4"> 
						{ curr_movie.Title } 
					</Typography>
				</div>
			</div>

			<div className={classes.image}>
				<img className={classes.poster} src={curr_movie.Poster} alt="curr_movie poster" />
				<div>
					{`Released...${curr_movie.Year}`}
				</div>
				<br />

				<div>
					{'Rated'}
					<h3> { curr_movie.Rated } </h3>
				</div>

				<div>
					<h6> { `imdbID: ${curr_movie.imdbID}` } </h6>
				</div>

			</div>

			<div className={classes.info}>

				<div>
					<h4> {'Genre:'} </h4>
					<p> {curr_movie.Genre} </p>
				</div>

				<div>
					<h4> { 'Director:' } </h4>
					<p> {curr_movie.Director} </p>
				</div>

				<div>
					<h4> {' Starring:' } </h4>
					<p> { actors } </p>
				</div>

			</div>

			<div className={classes.ratings}>
				<div>
					<h4> Critics Ratings: </h4>
					<p> { ratings } </p>
					<br />
				</div>
			</div>

			<div className={classes.plot}>
				<h4> Movie Plot: </h4>
				<p> { curr_movie.Plot } </p>
			</div>

		</div>
	);
}

const styles = theme => ({

	fav: {
		display: 'inherit',
		gridArea: 'fav',
		padding: theme.spacing.unit,
		placeContent: 'center',
	},

	image: {
		borderRight: '1px solid lightgrey',
		gridArea: 'image',
		padding: theme.spacing.unit,
	},

	info: {
		gridArea: 'info',
		padding: theme.spacing.unit * 2,
		textAlign: 'left',
	},

	main: {
		border: '1px solid lightgrey',
		display: 'inline-grid',
		gridTemplateAreas: `
      "titlebar titlebar"
      "image info"
      "plot plot"
      "ratings ratings"
    `,
		gridTemplateColumns: '2fr 3fr',
	},

	plot: {
		border: '1px solid lightgrey',
		gridArea: 'plot',
		padding: theme.spacing.unit,
		textAlign: 'left',
	},

	poster: {
		maxWidth: 125,
		padding: theme.spacing.unit,
	},

	ratingUnit: {
		display: 'grid',
		gridTemplateAreas: `
      "critic grade"
    `,
		gridTemplateColumns: '3rf 1fr',
	},

	ratings: {
		gridArea: 'ratings',
		padding: theme.spacing.unit,
		textAlign: 'left',
	},

	title: {
		fontColor: 'white',
		gridArea: 'title',
		padding: theme.spacing.unit,
	},

	titlebar: {
		display: 'inline-grid',
		backgroundColor: 'lightgrey',
		borderBottom: '1px solid lightgrey',
		fontColor: 'white',
		gridArea: 'titlebar',
		gridTemplateAreas: `
			"fav title"
		`,
		gridTemplateColumns: '1fr 9fr',
		padding: theme.spacing.unit,
	},


});

MovieDisplay.propTypes = propTypes;

export default withStyles(styles)(MovieDisplay);
