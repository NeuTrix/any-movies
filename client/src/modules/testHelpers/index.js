export { default as mockState } from './mockState';
