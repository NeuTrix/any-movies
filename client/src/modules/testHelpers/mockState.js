// present a mock state object for testing
// Need to consolidate this into a factory or other test helper file
const mockstate = {
	currMovie: {
		imdb_id: 'tt0078748',
		ratings: [{
			metacritic: 90,
			tomatoes: 10,
		}],
		title: 'Alien',
	},
	currUser: {
		id: 20,
		username: 'mickey333',
	},
	favourites: {},
};

export default mockstate;
