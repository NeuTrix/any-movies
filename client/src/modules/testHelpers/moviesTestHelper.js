// normalizr schema variables
import { normalize } from 'normalizr';
import { moviesListSchema } from '../movies/redux';

// test objects
// export to a faker or factory helper file
// const movie1 = [{
// 		title: 'Alien',
// 		imdb_id: 'tt0078748'
// }];

// const movie2 = [
// 	{ title: 'Star Wars',
// 			imdb_id: 'tt0076759'
// 	}
// ];

// normalized
const normed1 = normalize(data1, moviesListSchema);
const normed2 = normalize(data2, moviesListSchema);

// export test variables
export const movies1 = normed1.result;
export const dictionary1 = normed1.entities.movies;
export const movies2 = normed2.result;
export const dictionary2 = normed2.entities.movies;
