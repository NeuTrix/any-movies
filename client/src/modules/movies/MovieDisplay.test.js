import React from 'react';
import ReactDOM from 'react-dom';
// import { mockstate } from '../../testHelpers';
import { mockstate } from '../testHelpers';
import MovieDisplay from './MovieDisplay';
import renderer from 'react-test-renderer';

describe('The MovieDisplay component', () => {
	const { currMovie, currUser } = mockstate;
	const props = { currMovie, currUser };

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<MovieDisplay {...props} />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	describe('The MovieDisplay snapshot', () => {

		const component = renderer.create(
			<MovieDisplay { ...props } />
		);
		
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	});

});
