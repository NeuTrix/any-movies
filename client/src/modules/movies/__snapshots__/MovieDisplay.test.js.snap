// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="MovieDisplay-main-4"
>
  <div
    className="MovieDisplay-titlebar-10"
  >
    <div
      className="MovieDisplay-fav-1"
    >
      <button
        className="MuiButtonBase-root-19 MuiIconButton-root-13"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        size="large"
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-18"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-22 FavouriteButton-favourited-11"
            focusable="false"
            role="presentation"
            style={
              Object {
                "color": "black",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <g>
              <path
                d="M16.5 5c-1.54 0-3.04.99-3.56 2.36h-1.87C10.54 5.99 9.04 5 7.5 5 5.5 5 4 6.5 4 8.5c0 2.89 3.14 5.74 7.9 10.05l.1.1.1-.1C16.86 14.24 20 11.39 20 8.5c0-2-1.5-3.5-3.5-3.5z"
                opacity=".3"
              />
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-67"
        />
      </button>
    </div>
    <div
      className="MovieDisplay-title-9"
    >
      <h4
        className="MuiTypography-root-31 MuiTypography-h4-46"
      />
    </div>
  </div>
  <div
    className="MovieDisplay-image-2"
  >
    <img
      alt="currMovie poster"
      className="MovieDisplay-poster-6"
    />
    <div>
      Released...undefined
    </div>
    <br />
    <div>
      Rated
      <h3>
         
         
      </h3>
    </div>
    <div>
      <h6>
         
        imdbID: undefined
         
      </h6>
    </div>
  </div>
  <div
    className="MovieDisplay-info-3"
  >
    <div>
      <h4>
         
        Genre:
         
      </h4>
      <p>
         
         
      </p>
    </div>
    <div>
      <h4>
         
        Director:
         
      </h4>
      <p>
         
         
      </p>
    </div>
    <div>
      <h4>
         
         Starring:
         
      </h4>
      <p>
         
         
      </p>
    </div>
  </div>
  <div
    className="MovieDisplay-ratings-8"
  >
    <div>
      <h4>
         Critics Ratings: 
      </h4>
      <p>
         
         
      </p>
      <br />
    </div>
  </div>
  <div
    className="MovieDisplay-plot-5"
  >
    <h4>
       Movie Plot: 
    </h4>
    <p>
       
       
    </p>
  </div>
</div>
`;
